src/memory_prefetch.c

```diff
@@ -0,0 +1,401 @@
+                                                         start
+                                                           │
+                                                  ┌────────▼─────────┐
+                                       ┌─────────►│  PREFETCH_BUCKET ├────►────────┐
+                                       │          └────────┬─────────┘            no more tables -> done
+                                       |             bucket|found                  |
+                                       │                   |                       │
+        entry not found - goto next table         ┌────────▼────────┐              │
+                                       └────◄─────┤ PREFETCH_ENTRY  |              ▼
+                                    ┌────────────►└────────┬────────┘              │
+                                    |                 Entry│found                  │
+                                    │                      |                       │
+                                    |              ┌───────▼────────┐              │
+                                    │              | PREFETCH_KVOBJ |              ▼
+                                    │              └───────┬────────┘              │
+        kvobj not found - goto next entry                  |                       |
+                                    │          ┌───────────▼────────────┐          │
+                                    └──────◄───│    PREFETCH_VALDATA    │          ▼
+                                               └───────────┬────────────┘          │
+                                                           |                       │
+                                                 ┌───────-─▼─────────────┐         │
+                                                 │     PREFETCH_DONE     │◄────────┘
+                                                 └───────────────────────┘
+**********************************************************************************************************************/
+
+typedef void *(*GetValueDataFunc)(const void *val);
+
+typedef struct KeyPrefetchInfo {
+    PrefetchState state;      /* Current state of the prefetch operation */
+    HashTableIndex ht_idx;    /* Index of the current hash table (0 or 1 for rehashing) */
+    uint64_t bucket_idx;      /* Index of the bucket in the current hash table */
+    uint64_t key_hash;        /* Hash value of the key being prefetched */
+    dictEntry *current_entry; /* Pointer to the current entry being processed */
+    kvobj *current_kv;        /* Pointer to the kv object being prefetched */
+} KeyPrefetchInfo;
+
+/* PrefetchCommandsBatch structure holds the state of the current batch of client commands being processed. */
+typedef struct PrefetchCommandsBatch {
+    size_t cur_idx;                 /* Index of the current key being processed */
+    size_t keys_done;               /* Number of keys that have been prefetched */
```

Comment:
is this unused?

Reply:
@ShooterIT I have a question. cur_idx is used in memory_prefetch.c 
do you mean it is unused?

```
/* Prefetch the given pointer and move to the next key in the batch. */                               static inline void prefetchAndMoveToNextKey(void *addr) {
    redis_prefetch(addr);
    /* While the prefetch is in progress, we can continue to the next key */
    batch->cur_idx = (batch->cur_idx + 1) % batch->key_count;
}
```

Reply:
i think we can remove it, since it does not take real effect.